decidim-process-extended/lib/decidim/process/extended/process_serializer.rb

```diff
@@ -0,0 +1,157 @@
+        # Public: Initializes the serializer with a ParticipatoryProcess.
+        def initialize(process)
+          @process = process
+        end
+
+        # Public: Returns a hash with the serialized data.
+        def serialize
+          {
+            # Process Information
+            id: process.id,
+            title_ca: process.title['ca'],
+            slug: process.slug,
+            short_description_ca: process.short_description['ca'],
+            process_type: process_type,
+            scope_id: process.scope&.id,
+            scope_name_ca: process.scope&.name.try(:[], 'ca'),
+            department_id: process.area&.id,
+            department_name_ca: process.area&.name.try(:[], 'ca'),
+            participatory_space: process.participatory_process_group&.name.try(:[], 'ca')&.downcase,
+            normative_type_id: process.scope&.scope_type&.id,
+            normative_type_name_ca: process.scope&.name.try(:[], 'ca'),
+            duration_days: duration_days,
+            start_date: process.start_date,
+            end_date: process.end_date,
+            cost: process.cost,
+            has_record: process.has_summary_record,
+            facilitators: process.facilitators,
+            promoting_unit: process.promoting_unit,
+            total_num_participants: meetings_num_participants + proposals_num_authors,
+            total_num_entities: meetings_num_entities + proposals_num_entities,
+            # Related Resources: Proposals
+            proposals_num_authors: proposals_num_authors,
+            proposals_num_author_entities: proposals_num_entities,
+            total_num_proposals: proposals.count,
+            total_accepted_proposals: proposals.accepted.count,
+            total_rejected_proposals: proposals.rejected.count,
+            total_evaluating_proposals: proposals.evaluating.count,
+            proposals_num_proposals: proposals_no_meeting.except_withdrawn.count,
+            # Related Resources: Meetings
+            meetings_num_participants: meetings_num_participants,
+            meetings_num_entities: meetings_num_entities,
+            total_num_meetings: meetings.count,
+            meetings_num_proposals: proposals.where(created_in_meeting: true).count,
+            # Related Resources: Debates
+            debates_num_debates: debates.count,
+            # Related Resources: Assemblies
+            has_related_assembly: assembly.present?,
+            related_assembly_name_ca: assembly&.title.try(:[], 'ca')
+          }
+        end
+
+        private
+
+        attr_reader :process
+
+        def process_type
+          types = []
+          types << :virtual if proposals.count.positive?
+          types << :presencial if meetings.count.positive?
+          types.size > 1 ? :ambdos : types.first
+        end
+
+        def duration_days
+          return unless process.start_date.present? && process.end_date.present?
+
+          (process.start_date...process.end_date).count
```

Comment:
diria que és més performant el format del meu PR: `(process.end_date-process.start_date).days`

Reply:
`(process.end_date-process.start_date).days` [returns a fraction](https://stackoverflow.com/a/20862867). Since I had to call `.to_i` at the end I opted for [this solution](https://stackoverflow.com/a/40131738), which seemed more elegant.

Reply:
But both seem fine to me. You can choose if you want:
`(process.end_date - process.start_date).to_i` 
`(process.start_date...process.end_date).count`

Reply:
I'm not sure but have the intuition that using a Range will require to iterate while subtracting is a single operation for sure.